src/controllers/deploy.go

```diff
@@ -99,6 +104,27 @@ func internaldeploy(a *history.ActionInstance) ([]byte, error) {
 }
 
 // GetDeployArgs - Get arguments to pass to deploy script as a string array
-func GetDeployArgs(repoURL string, branch string, server string, subdomain string, access string) []string {
-	return []string{"-n", "-u", repoURL, "-b", branch, "-m", server, "-s", subdomain, "-a", access}
+func GetDeployArgs(repoURL string, branch string, server string, subdomain string, access string, kwargs map[string]bool) []string {
+	args := []string{"-n", "-u", repoURL, "-b", branch, "-m", server, "-s", subdomain, "-a", access}
+	if v, ok := kwargs["redeploy"]; ok && v {
+		args = append(args, "-x")
+	}
+	if v, ok := kwargs["restart"]; ok && v {
+		args = append(args, "-r")
+	}
```

Comment:
I think that kwargs is unnecessarily complicating the flow. keep it as it was before and handle the new args in their respective routines